diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index d36eee192..98203a394 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -5,17 +5,18 @@ function(dissolve_system_test test_name input_file count)
   message(STATUS "... System test '${test_name}' in directory '${CMAKE_CURRENT_LIST_DIR}'")
   if(PARALLEL)
     foreach(nproc 1 2)
-      set(test_target_executable ${CMAKE_BINARY_DIR}/bin/${target_name})
+      set(test_target_executable ${target_name})
+      string (REPLACE  ";" " " vargs "${ARGN}")
       add_test(
         NAME ${CMAKE_CURRENT_LIST_DIR}-${test_name}-${nproc}
-        COMMAND ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${nproc} ${test_target_executable} -n ${count} -x ${input_file} ${ARGN}
+        COMMAND sh -c "echo ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${nproc} `find ../../build/nix/store -name ${test_target_executable}` -n ${count} -x ${input_file} ${vargs} && ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${nproc} `find ../../build/nix/store -name ${test_target_executable}` -n ${count} -x ${input_file} ${vargs}"
         WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
       )
       set_property(TEST ${CMAKE_CURRENT_LIST_DIR}-${test_name}-${nproc} PROPERTY LABELS ${CMAKE_CURRENT_LIST_DIR})
       set_property(TEST ${CMAKE_CURRENT_LIST_DIR}-${test_name}-${nproc} PROPERTY PROCESSORS ${nproc})
     endforeach()
   else()
-    set(test_target_executable ${CMAKE_BINARY_DIR}/bin/${target_name})
+    set(test_target_executable ${target_name})
     add_test(
       NAME ${CMAKE_CURRENT_LIST_DIR}-${test_name}
       COMMAND ${test_target_executable} -n ${count} -x ${input_file} ${ARGN}
