diff --git a/CMakeLists.txt b/CMakeLists.txt
index a9e748fc6..5f26682d0 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,9 +23,7 @@ cmake_policy(SET CMP0020 NEW)
 # Request C++17 Standard Support (Required for std::optional and auto types in lambda parameters)
 set(CMAKE_CXX_STANDARD 17)
 
-# Include Conan Packages
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
+find_package(fmt REQUIRED)
 
 # Catch defines
 if(PARALLEL)
@@ -315,7 +313,7 @@ target_link_libraries(
          ${MODULENOGUI_LINK_LIBS} ${NO_WHOLE_ARCHIVE_FLAG}
   PRIVATE # External libs
           antlr4-runtime ${EXTRA_LINK_LIBS} ${THREADING_LINK_LIBS}
-  INTERFACE CONAN_PKG::fmt CONAN_PKG::cli11
+  INTERFACE fmt::fmt cli11
 )
 
 # Build GUI target executables
@@ -347,6 +345,6 @@ if(GUI)
             ${FREETYPE_LIBRARIES}
             ${EXTRA_LINK_LIBS}
             ${THREADING_LINK_LIBS}
-    INTERFACE CONAN_PKG::fmt CONAN_PKG::cli11 CONAN_PKG::pugixml
+    INTERFACE fmt::fmt cli11 pugixml
   )
 endif(GUI)
diff --git a/src/base/CMakeLists.txt b/src/base/CMakeLists.txt
index 2f27c755e..85e111b70 100644
--- a/src/base/CMakeLists.txt
+++ b/src/base/CMakeLists.txt
@@ -29,4 +29,4 @@ add_library(
 )
 
 include_directories(base PRIVATE ${PROJECT_SOURCE_DIR}/src)
-target_link_libraries(base PUBLIC CONAN_PKG::fmt ${THREADING_LINK_LIBS})
+target_link_libraries(base PUBLIC fmt::fmt ${THREADING_LINK_LIBS})
diff --git a/src/main/CMakeLists.txt b/src/main/CMakeLists.txt
index 5e5d464ac..61b4e7185 100644
--- a/src/main/CMakeLists.txt
+++ b/src/main/CMakeLists.txt
@@ -28,4 +28,4 @@ add_library(
 )
 
 include_directories(main PRIVATE ${PROJECT_SOURCE_DIR}/src)
-target_link_libraries(main PRIVATE CONAN_PKG::fmt CONAN_PKG::cli11)
+target_link_libraries(main PRIVATE fmt::fmt cli11)
diff --git a/unit/CMakeLists.txt b/unit/CMakeLists.txt
index 6cc9bf4bc..ead38c01c 100644
--- a/unit/CMakeLists.txt
+++ b/unit/CMakeLists.txt
@@ -23,7 +23,7 @@ function(dissolve_unit_test src_file)
     ${TEST_NAME}
     PUBLIC ${WHOLE_ARCHIVE_FLAG} ${BASIC_LINK_LIBS} ${MODULENOGUI_LINK_LIBS} ${NO_WHOLE_ARCHIVE_FLAG}
     PRIVATE antlr4-runtime ${EXTRA_LINK_LIBS} GTest::gtest_main
-    INTERFACE CONAN_PKG::fmt CONAN_PKG::cli11
+    INTERFACE fmt::fmt cli11
   )
 
   if(GUI)
diff --git a/src/procedure/CMakeLists.txt b/src/procedure/CMakeLists.txt
index bc806b3e9..ea2e3ef9f 100644
--- a/src/procedure/CMakeLists.txt
+++ b/src/procedure/CMakeLists.txt
@@ -3,4 +3,4 @@ add_subdirectory(nodes)
 add_library(procedure procedure.cpp nodevalue.cpp procedure.h nodevalue.h)
 
 include_directories(procedure PRIVATE ${PROJECT_SOURCE_DIR}/src)
-target_link_libraries(procedure PRIVATE CONAN_PKG::fmt)
+target_link_libraries(procedure PRIVATE fmt::fmt)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a9e748fc6..8e094e414 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,8 @@ endif(PARALLEL)
 option(MULTI_THREADING "Enable threading using tbb" ON)
 if(MULTI_THREADING)
   add_definitions("-DMULTITHREADING")
-  set(THREADING_LINK_LIBS ${THREADING_LINK_LIBS} CONAN_PKG::tbb)
+  set(THREADING_LINK_LIBS ${THREADING_LINK_LIBS} ${TBB_LIBRARIES})
+  include_directories(${TBB_INCLUDE})
 endif(MULTI_THREADING)
 
 # Add local Modules dir to cmake search path
